libsignal-service/src/service_address.rs

```diff
@@ -13,64 +15,124 @@ pub enum ParseServiceAddressError {
     NoUuid,
 }
 
-#[derive(Clone, Copy, Debug, Eq, PartialEq, Serialize, Deserialize)]
+#[derive(Clone, Copy, Debug, Eq, PartialEq, Serialize, Deserialize, Hash)]
 pub struct ServiceAddress {
     pub uuid: Uuid,
+    pub identity: ServiceIdType,
 }
 
 impl ServiceAddress {
     pub fn to_protocol_address(
         &self,
         device_id: impl Into<DeviceId>,
     ) -> ProtocolAddress {
-        ProtocolAddress::new(self.uuid.to_string(), device_id.into())
+        match self.identity {
+            ServiceIdType::AccountIdentity => {
+                ProtocolAddress::new(self.uuid.to_string(), device_id.into())
+            },
+            ServiceIdType::PhoneNumberIdentity => ProtocolAddress::new(
+                format!("PNI:{}", self.uuid),
```

Comment:
We chop this prefix off in Whisperfish in the protocol store, right?

Reply:
It looks like we don't - but that's a discussion to be had in the Whisperfish MR.